tests/perf-test/SidebarUtils.perf-test.ts

```diff
@@ -24,60 +27,60 @@ afterEach(() => {
+const getMockedReports = (length = 500) =>
+    createCollection<Report>(
+        (item) => `report_${item.reportID}`,
+        (index) => createRandomReport(index),
+        length,
+    ) as Partial<Report>;
 
-        return {[reportKey]: report};
-    });
-
-    return Object.assign({}, ...mockReports) as Partial<Report>;
-};
-
-const getMockedPoliciesMap = (length = 100) => {
-    const mockPolicies = Array.from({length}, (v, i) => {
-        const policyID = i + 1;
-        const policyKey = `${ONYXKEYS.COLLECTION.POLICY}${policyID}`;
-        const policy = LHNTestUtils.getFakePolicy();
-
-        return {[policyKey]: policy};
-    });
-
-    return Object.assign({}, ...mockPolicies) as Record<string, Policy>;
-};
+const reportActions = createCollection<ReportAction>(
+    (item) => `${item.reportActionID}`,
+    (index) => createRandomReportAction(index),
+);
 
-const mockedResponseMap = getMockedReportsMap(10000);
+const mockedResponseMap = getMockedReports(10000);
 
 test('getOptionData on 10k reports', async () => {
-    const report = LHNTestUtils.getFakeReportWithPolicy([1, 2], 1, true) as Report;
-    const reportActions = ReportTestUtils.getMockedReportActionsMap();
+    const report = createRandomReport(1);
     const personalDetails = LHNTestUtils.fakePersonalDetails;
     const preferredLocale = 'en';
-    const policy = LHNTestUtils.getFakePolicy() as Policy;
-    const reportAction = ReportTestUtils.getFakeReportAction(1, 'ADDCOMMENT') as unknown as ReportAction;
+    const policy = createRandomPolicy(1);
+    const parentReportAction = createRandomReportAction(1);
 
     Onyx.multiSet({
         ...mockedResponseMap,
     });
 
     await waitForBatchedUpdates();
-    await measureFunction(() => SidebarUtils.getOptionData(report, reportActions, personalDetails, preferredLocale, policy, reportAction), {runs: 20});
+    await measureFunction(() => SidebarUtils.getOptionData(report, reportActions, personalDetails, preferredLocale, policy, parentReportAction), {runs: 20});
 });
 
 test('getOrderedReportIDs on 10k reports', async () => {
     const currentReportId = '1';
-    const allReports = getMockedReportsMap() as Record<string, Report>;
+    const allReports = getMockedReports() as Record<string, Report>;
     const betas = [CONST.BETAS.DEFAULT_ROOMS, CONST.BETAS.POLICY_ROOMS];
-    const policies = getMockedPoliciesMap();
-    const reportActions = ReportTestUtils.getMockedReportActionsMap();
+
+    const policies = createCollection<Policy>(
+        (item) => `policy_${item.id}`,
+        (index) => createRandomPolicy(index),
+    );
+
+    const allReportActions = Object.values(reportActions).map((reportAction) => ({
+        errors: reportAction.errors ?? [],
+        message: [
+            {
+                moderationDecision: {
+                    decision: reportAction.message?.[0]?.moderationDecision?.decision,
+                },
+            },
+        ],
+    })) as unknown as Record<string, ReportAction[]>;
```

Comment:
are typing completely gone at this stage? should be possible to rely on infer

Reply:
I had to fix this mock as it was incorrect. However, due to the narrowing of the scope of reportActions, additional typing is needed here.